includes/DataDumpPager.php

```diff
@@ -2,185 +2,201 @@
 use MediaWiki\Pager\TablePager;
 use MediaWiki\Permissions\PermissionManager;
-use MediaWiki\Title\Title;
+use MediaWiki\SpecialPage\SpecialPage;
 use Miraheze\DataDump\Jobs\DataDumpGenerateJob;
 use PermissionsError;
+use stdClass;
+use Wikimedia\Rdbms\IConnectionProvider;
 
 class DataDumpPager extends TablePager {
 
-	/** @var Config */
-	private $config;
-
-	/** @var Title */
-	private $pageTitle;
-
-	/** @var PermissionManager */
-	private $permissionManager;
-
-	public function __construct( IContextSource $context, $pageTitle ) {
-		$this->setContext( $context );
-
-		$this->mDb = MediaWikiServices::getInstance()
-			->getDBLoadBalancer()
-			->getMaintenanceConnectionRef( DB_PRIMARY );
-
-		if ( !$this->getRequest()->getVal( 'sort', null ) ) {
-			$this->mDefaultDirection = true;
-		}
-
-		parent::__construct( $context );
-
-		$this->config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'DataDump' );
-		$this->pageTitle = $pageTitle;
-		$this->permissionManager = MediaWikiServices::getInstance()->getPermissionManager();
+	private Config $config;
+	private JobQueueGroupFactory $jobQueueGroupFactory;
+	private LinkRenderer $linkRenderer;
+	private PermissionManager $permissionManager;
+
+	public function __construct(
+		Config $config,
+		IContextSource $context,
+		IConnectionProvider $connectionProvider,
+		JobQueueGroupFactory $jobQueueGroupFactory,
+		LinkRenderer $linkRenderer,
+		PermissionManager $permissionManager
+	) {
+		parent::__construct( $context, $linkRenderer );
+
+		$this->mDb = $connectionProvider->getPrimaryDatabase();
+
+		$this->config = $config;
+		$this->jobQueueGroupFactory  = $jobQueueGroupFactory;
+		$this->linkRenderer = $linkRenderer;
+		$this->permissionManager = $permissionManager;
 	}
 
-	public function getFieldNames() {
-		static $headers = null;
-
-		$headers = [
-			'dumps_timestamp' => 'datadump-table-header-date',
-			'dumps_filename'  => 'datadump-table-header-name',
-			'dumps_type'      => 'datadump-table-header-type',
-			'dumps_size'      => 'datadump-table-header-size',
-			'dumps_status'    => 'datadump-table-header-status',
-			'dumps_delete'    => 'datadump-table-header-delete',
+	/** @inheritDoc */
+	public function getFieldNames(): array {
+		return [
+			'dumps_timestamp' => $this->msg( 'datadump-table-header-date' )->text(),
+			'dumps_filename' => $this->msg( 'datadump-table-header-name' )->text(),
+			'dumps_type' => $this->msg( 'datadump-table-header-type' )->text(),
+			'dumps_size' => $this->msg( 'datadump-table-header-size' )->text(),
+			'dumps_status' => $this->msg( 'datadump-table-header-status' )->text(),
+			'dumps_delete' => $this->msg( 'datadump-table-header-delete' )->text(),
 		];
-
-		foreach ( $headers as &$msg ) {
-			$msg = $this->msg( $msg )->text();
-		}
-
-		return $headers;
 	}
 
-	public function formatValue( $name, $value ) {
-		$row = $this->mCurrentRow;
+	/** @inheritDoc */
+	public function formatValue( $name, $value ): string {
```

Comment:
Type hint?

Reply:
Can't type hint an override function when the source isn't as then it throws errors.